### ISSUE TITLE
Unsafe Style/IfUnlessModifier Autocorrection for Code Using `defined?`

### ISSUE THREAD

**Issue by chadrschroeder** [rubocop/rubocop#11040]
Say that I have code like:

```ruby
# Original Code
unless defined?(value)
  value = 'default'
end
```

This is reported as a `Style/IfUnlessModifier` failure.  If I autocorrect it, it changes to:

```ruby
# Autocorrected Code
value = 'default' unless defined?(value)
```

If `value` isn't defined, the original code will set it to `'default'` while the autocorrected code will set it to `nil`.

I usually try to avoid using `defined?` altogether.  But the example above is similar to what I was seeing in some legacy code and I wanted to document it.

Often code like this can be rewritten to use `||=`, as long a `false` value doesn't ever need to override the default.  Or it could be rewritten to use an additional `value_defined` variable that is set before the assignment of `value`.

I'm not sure what an autocorrection could be for this situation.  Maybe RuboCop shouldn't do anything here?

## RuboCop version

```
$ [bundle exec] rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.0.3) [x86_64-darwin20]
  - rubocop-rails 2.16.1
```
